app/stores/interactive-filters.tsx

```diff
@@ -141,80 +142,62 @@ type InteractiveFiltersContextValue = [
   StoreApi<State>,
 ];
 
-export type SharedFilter = {
-  type: "timeRange" | "dataFilters";
-  value?: NonNullable<InteractiveFiltersConfig>["timeRange"];
-  iri: string;
-};
+export type SharedFilter =
+  | ({
+      type: "timeRange";
+    } & NonNullable<InteractiveFiltersConfig>["timeRange"])
+  | ({
+      type: "dataFilters";
+    } & NonNullable<InteractiveFiltersConfig>["dataFilters"]);
 
-type SharedFilters = SharedFilter[];
+export type PotentialSharedFilter = Pick<SharedFilter, "type" | "componentIri">;
 
 const InteractiveFiltersContext = createContext<
   | {
       stores: Record<ChartConfig["key"], InteractiveFiltersContextValue>;
-      sharedFilters: SharedFilters;
+      potentialSharedFilters: PotentialSharedFilter[];
+      sharedFilters: SharedFilter[];
     }
   | undefined
 >(undefined);
 
 /**
  * Returns filters that are shared across multiple charts.
  */
-export const getSharedFilters = (
-  state: ConfiguratorStateWithChartConfigs
-): SharedFilters => {
-  const chartConfigs = state.chartConfigs;
-
-  // We show shared filters only when the chart is not being configured
-  if (state.state === "CONFIGURING_CHART") {
-    return [];
-  }
-
-  // Shared filters are only relevant for dashboard layout
-  if (state.layout.type !== "dashboard") {
-    return [];
-  }
-
-  if (chartConfigs.length < 2) {
-    return [];
-  }
-
-  const allActiveFilters = chartConfigs.flatMap((c) => {
-    const interactiveFiltersConfig = c.interactiveFiltersConfig;
-    if (!interactiveFiltersConfig) {
-      return [];
-    }
-
-    const { timeRange } = interactiveFiltersConfig;
-    return [{ type: "timeRange" as const, value: timeRange }].filter(
-      (x) => x.value.active
+export const getPotentialSharedFilters = (
+  chartConfigs: ChartConfig[]
+): PotentialSharedFilter[] => {
+  const temporalDimensions = chartConfigs.flatMap((c) => {
+    const chartSpec = getChartSpec(c);
+    const temporalEncodings = chartSpec.encodings.filter((x) =>
+      x.componentTypes.some((x) => x === "TemporalDimension")
     );
-  });
-
-  const sharedFilters = Object.entries(
-    // TODO implement recognizing shared filters across joined by dimensions
-    groupBy(allActiveFilters, (x) => `${x.type} - ${x.value.componentIri}`)
-  ).filter(([_iri, filters]) => filters.length > 1);
-
-  return sharedFilters.map(([_iri, filters]) => ({
-    iri: filters[0].value.componentIri,
-    type: filters[0].type,
-    value: (() => {
-      const type = filters[0].type;
-      switch (type) {
-        case "timeRange":
+    const chartTemporalDimensions = temporalEncodings
+      .map((x) => {
+        // @ts-ignore
+        const field = x.field in c.fields ? c.fields[x.field] : undefined;
+        if (field && "componentIri" in field) {
```

Comment:
Nit: Might be a bit more clear with e.g. `config` instead of `c` and `encoding` instead of `x`. But might also be clear in the code, just something I wasn't sure in GitHub interface when reviewing :)